testing/test_deploy.py

```diff
@@ -158,20 +158,49 @@ def setup(args):
   if args.cluster:
     util.configure_kubectl(args.project, args.zone, args.cluster)
 
-  apply_command = ["ks", "apply", "default", "-c", "kubeflow-core",]
-
-  util.run(apply_command, cwd=app_dir)
-
-  # Verify that the TfJob operator is actually deployed.
-  tf_job_deployment_name = "tf-job-operator"
-  logging.info("Verifying TfJob controller started.")
-  util.wait_for_deployment(api_client, namespace.metadata.name,
-                           tf_job_deployment_name)
-
-  # Verify that JupyterHub is actually deployed.
-  jupyter_name = "tf-hub"
-  logging.info("Verifying TfHub started.")
-  util.wait_for_statefulset(api_client, namespace.metadata.name, jupyter_name)
+  if args.deploy_core:
+    apply_command = ["ks", "apply", "default", "-c", "kubeflow-core",]
+
+    util.run(apply_command, cwd=app_dir)
+
+    # Verify that the TfJob operator is actually deployed.
+    tf_job_deployment_name = "tf-job-operator"
+    logging.info("Verifying TfJob controller started.")
+    util.wait_for_deployment(api_client, namespace.metadata.name,
+                             tf_job_deployment_name)
+
+    # Verify that JupyterHub is actually deployed.
+    jupyter_name = "tf-hub"
+    logging.info("Verifying TfHub started.")
+    util.wait_for_statefulset(api_client, namespace.metadata.name, jupyter_name)
+
+  if args.deploy_tf_serving:
+    logging.info("Deploying tf-serving.")
+    model_server_image = (args.model_server_image + "-" +
+                          os.getenv("JOB_TYPE", "") + "-" +
+                          os.getenv("PULL_BASE_SHA", ""))
+    util.run(["ks", "generate", "tf-serving", "modelServer",
+              "--name=inception",
+              "--namespace=" + namespace.metadata.name,
+              "--model_path=gs://kubeflow-models/inception",
+              "--model_server_image=" + model_server_image], cwd=app_dir)
+
+    apply_command = ["ks", "apply", "default", "-c", "modelServer",]
+    util.run(apply_command, cwd=app_dir)
+
+    core_api = k8s_client.CoreV1Api(api_client)
+    deploy = core_api.read_namespaced_service(
+        "inception", namespace.metadata.name)
+    cluster_ip = deploy.spec.cluster_ip
+
+    util.wait_for_deployment(api_client, namespace.metadata.name, "inception")
+    logging.info("Verified TF serving started.")
+
+    if args.test_inception and args.inception_client_image:
+      util.run(["gcloud", "docker", "--", "pull", args.inception_client_image])
```

Comment:
Why are you pulling the inception client image?  Why not
1. Run the test as a K8s job?
or 
2. Bake any required dependencies into the test image:
gcr.io/kubeflow-ci/test-worker:latest

Reply:
Actually any reason you can't just have an Argo step that uses the inception_client_image to run any code?

Are we using the inception model for testing?